tests/DocumentTest.php

```diff
@@ -418,6 +461,252 @@ public function testFromFile()
+            ),
+            $creative->getTrackingEvents()
+        );
+
+        // test with no tracking events
+        /** @var InLine $adSection */
+        $inlineAd = $adSections[1];
+        $creatives = $inlineAd->getCreatives();
+        /** @var \Sokil\Vast\Creative\AbstractLinearCreative $creative */
+        $creative = $creatives[0];
+        self::assertEquals(
+            array(),
+            $creative->getTrackingEvents()
+        );
+
+        // test with empty tracking events tag
+        /** @var InLine $adSection */
+        $inlineAd = $adSections[2];
+        $creatives = $inlineAd->getCreatives();
+        /** @var \Sokil\Vast\Creative\AbstractLinearCreative $creative */
+        $creative = $creatives[0];
+        self::assertEquals(
+            array(),
+            $creative->getTrackingEvents()
+        );
+
+        // test with multiurls and w/o attribute
+        /** @var InLine $adSection */
+        $inlineAd = $adSections[3];
+        $creatives = $inlineAd->getCreatives();
+        /** @var \Sokil\Vast\Creative\AbstractLinearCreative $creative */
+        $creative = $creatives[0];
+        self::assertEquals(
+            array(
+                'start' => array('http://ad.server.com/trackingevent/start', ),
+                'pause' => array('http://ad.server.com/trackingevent/pause', 'http://ad.server.com/trackingevent/pauseAlt'),
+            ),
+            $creative->getTrackingEvents()
+        );
+    }
+
+    /**
+     * @expectedException        \Exception
+     * @expectedExceptionMessage Ad type WrongAdSection not supported
+     */
+    public function testFromBrokenFile()
+    {
+        $factory = new Factory();
+        $document = $factory->fromFile(__DIR__ . '/data/vastBroken.xml');
+        $document->getAdSections();
+    }
+
+    /**
+     * @expectedException        \Exception
+     * @expectedExceptionMessage Wrong event "BrokenEvent" specified
+     */
+    public function testBrokenEvent()
+    {
+        $factory = new Factory();
+        $document = $factory->create('2.0');
+
+        // insert Ad section
+        $ad1 = $document
+            ->createInLineAdSection()
+            ->setId('ad1')
+            ->setAdSystem('Ad Server Name')
+            ->setAdTitle('Ad Title')
+            ->addImpression('http://ad.server.com/impression', 'imp1');
+
+        // create creative for ad section
+        $ad1
+            ->createLinearCreative()
+            ->addTrackingEvent('BrokenEvent', 'http://ad.server.com/trackingevent/BrokenEvent')
+        ;
+    }
+
+    /**
+     * @expectedException        \InvalidArgumentException
+     * @expectedExceptionMessage Unknown scalar node NonExistentTag
+     */
+    public function testBrokenScalarNodeValue()
+    {
+        $class = new \ReflectionClass('Sokil\Vast\Ad\InLine');
+        $method = $class->getMethod('getScalarNodeValue');
+        $method->setAccessible(true);
```

Comment:
bad idea. Reflection broke incapsulation. If you can call getScalarNodeValue with invalid name in some public method, test this public method. If no such method  - its nothing to test. I think its just for 100% coverage ))))

Coverage do not mean that your code 100% valid. It means that you call 100% of lines in your test, and nothing more. You can write test that call anything and do no assertions and your coverage will be 100% but code not tested.